inspirehep/modules/workflows/workflows/article.py

```diff
@@ -317,7 +331,9 @@
 
 
 STORE_RECORD = [
-    store_record,
+    do_not_repeat('store_record')(
```

Comment:
I am not sure this is correct. What you really want to avoid is minting a new recid, but here you avoid saving it altogether. This doesn't matter currently as we still have legacy that will overwrite the record anyway, but is not really robust.

The better solution is probably to write the minted recid also somewhere to `extra_data` (wrapping https://github.com/inspirehep/inspire-next/blob/cd0ef5d5309f63ef4ce4f5f69604638ed55a7cf4/inspirehep/modules/workflows/tasks/upload.py in a `try...finally` to make sure that always happens, even if the record creation raises) and have a mechanism to make that survive restarting, generalizing `skipped_steps` to be able to store arbitrary data.

Reply:
After some discussions IRL, we have decided to not skip `store_data` since it should be harmless to run it again in case of failures or restarts.